src/components/containers/TraceAccordion.js

```diff
@@ -71,28 +81,54 @@ class TraceAccordion extends Component {
   }
 
   renderUngroupedTraceFolds() {
-    return this.filteredTraces.map((d, i) => (
-      <TraceFold
-        key={i}
-        traceIndexes={[d.index]}
-        canDelete={this.props.canAdd}
-        fullDataArrayPosition={[this.filteredTracesIndexes[i]]}
-      >
-        {this.props.children}
-      </TraceFold>
-    ));
+    if (this.filteredTraces.length) {
+      return this.filteredTraces.map((d, i) => (
+        <TraceFold
+          key={i}
+          traceIndexes={[d.index]}
+          canDelete={this.props.canAdd}
+          fullDataArrayPosition={[d._expandedIndex]}
+        >
+          {this.props.children}
+        </TraceFold>
+      ));
+    }
+    return null;
   }
 
   renderTraceFolds() {
-    return this.filteredTraces.map((d, i) => (
-      <TraceFold
-        key={i}
-        traceIndexes={[this.filteredTracesIndexes[i]]}
-        canDelete={this.props.canAdd}
-      >
-        {this.props.children}
-      </TraceFold>
-    ));
+    if (this.filteredTraces.length) {
+      return this.filteredTraces.map((d, i) => (
+        <TraceFold
+          key={i}
+          traceIndexes={[this.filteredTracesDataIndexes[i]]}
+          canDelete={this.props.canAdd}
+        >
+          {this.props.children}
+        </TraceFold>
+      ));
+    }
+    return null;
```

Comment:
why did you change this to `null` from `[]`?

Reply:
just to avoid the `.length` check below?

Reply:
yeah... that was just when I was trying things to get the breaking test to work..
and I thought, well, since this function is called `renderTraceFolds` then it felt more reacty to return null if those shouldn't display..

Reply:
hehe I don't think that feels more reacty at all :) but ok.